backend/Social/SocialAPI.py

```diff
@@ -410,11 +410,23 @@ def getPostById(userID):
             "status": "success",
             "data": json.dumps(list(data), default=lambda o: str(o))
         }
-        return make_response(response, 200)
+
+        if db.Profiles.find_one({"userID": userID}) != None:
+            return make_response(response, 200)
+        
+        if db.Profiles.find_one({"userID": userID}) == None:
+            return make_response({"status": "failed", "message": "userID does not exist"}), 400   
+    except TypeError:
+        return make_response({"status": "failed", "message": "Expected a string input"}), 400
+    except ConnectionRefusedError as e:
+        print(e)
+        return make_response({"status": "failed", "message": "Connection refused"}), 408
```

Comment:
Connection refused occurs when mongo is not avail. Status should be 500.

BUT we should not open this detail to user, just return error as generic 500 under thhe general catch